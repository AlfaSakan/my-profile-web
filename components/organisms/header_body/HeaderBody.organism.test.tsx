import { fireEvent, render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import HeaderBody from './HeaderBody.organism';

describe('HeaderBody component test', () => {
  it('should render correctly', () => {
    const onClickMenu = jest.fn();

    const tree = renderer
      .create(
        <HeaderBody
          isFocus
          name="test"
          onClickMenu={onClickMenu}
          status="status"
        />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should called onClickMenu', () => {
    const onClickMenu = jest.fn();

    render(
      <HeaderBody
        isFocus
        name="test"
        onClickMenu={onClickMenu}
        status="status"
      />
    );

    fireEvent.click(screen.getByTestId('header-body-menu'));

    expect(onClickMenu).toHaveBeenCalled();
  });
});
