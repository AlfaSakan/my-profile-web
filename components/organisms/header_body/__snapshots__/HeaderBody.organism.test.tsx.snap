// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderBody component test should render correctly 1`] = `
<div
  className="flex items-center bg-secondary p-3 justify-between h-16"
>
  <div
    className="flex items-center"
  >
    <div
      className="mr-2 sm:hidden cursor-pointer"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
        />
      </svg>
    </div>
    <div
      className="rounded-full bg-gray-300 mr-4 flex items-center justify-center overflow-hidden h-10 w-10"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root"
        focusable="false"
        style={
          Object {
            "fill": "white",
          }
        }
        viewBox="0 0 24 24"
      >
        <path
          d="M16 11c1.66 0 2.99-1.34 2.99-3S17.66 5 16 5c-1.66 0-3 1.34-3 3s1.34 3 3 3zm-8 0c1.66 0 2.99-1.34 2.99-3S9.66 5 8 5C6.34 5 5 6.34 5 8s1.34 3 3 3zm0 2c-2.33 0-7 1.17-7 3.5V19h14v-2.5c0-2.33-4.67-3.5-7-3.5zm8 0c-.29 0-.62.02-.97.05 1.16.84 1.97 1.97 1.97 3.45V19h6v-2.5c0-2.33-4.67-3.5-7-3.5z"
        />
      </svg>
    </div>
    <div
      className="flex flex-col"
    >
      <p>
        test
      </p>
      <p
        className="text-xs mt-1"
      >
        status
      </p>
    </div>
  </div>
  <div
    className="flex items-center"
  >
    <div
      className="sm:mr-5"
    >
      <div
        className="cursor-pointer"
        onClick={[Function]}
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          style={
            Object {
              "fill": "#2C3333",
              "fontSize": 20,
            }
          }
          viewBox="0 0 24 24"
        >
          <path
            d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
          />
        </svg>
      </div>
    </div>
    <div
      className="cursor-pointer hidden sm:flex"
      data-testid="header-body-menu"
      onClick={[MockFunction]}
    >
      <button
        className="flex items-center justify-center w-32 h-12 rounded-lg bg-primary"
        style={
          Object {
            "height": 30,
            "width": 100,
          }
        }
      >
        <p>
          Keluar
        </p>
      </button>
    </div>
  </div>
</div>
`;
